src/vs/workbench/services/keybinding/browser/keybindingService.ts

```diff
@@ -535,6 +535,30 @@ export class WorkbenchKeybindingService extends AbstractKeybindingService {
 			return false;
 		}
 		const code = ScanCodeUtils.toEnum(event.code);
+
+		const NUMPAD_PRINTABLE_SCANCODES = [
+			ScanCode.NumpadDivide,
+			ScanCode.NumpadMultiply,
+			ScanCode.NumpadSubtract,
+			ScanCode.NumpadAdd,
+			ScanCode.Numpad1,
+			ScanCode.Numpad2,
+			ScanCode.Numpad3,
+			ScanCode.Numpad4,
+			ScanCode.Numpad5,
+			ScanCode.Numpad6,
+			ScanCode.Numpad7,
+			ScanCode.Numpad8,
+			ScanCode.Numpad9,
+			ScanCode.Numpad0,
+			ScanCode.NumpadDecimal
+		];
+		const immutableScanCode = IMMUTABLE_KEY_CODE_TO_CODE[event.keyCode];
+		if (NUMPAD_PRINTABLE_SCANCODES.indexOf(code) >= 0 && code === immutableScanCode) {
+			// code === immutableScanCode when NumLock is enabled
+			return true;
```

Comment:
We should rather move the condition to the end of the method and ask the opposite question.